src/calculateAverage.js

```diff
@@ -37,7 +37,32 @@ function applyCalculateAverage() {
    * @param {Object} options - optional
    */
   [].__proto__.calculateAverage = function(options) {
-    // write code here
+    if (options === undefined) {
+      return this.reduce((acum, num) => acum + num, 0) / this.length;
+    } else {
+      for (const key in options) {
+        if (key === 'propertyName') {
+          let counter = 0;
+          const sumOptions = options[key] === undefined
+            ? undefined
+            : this.reduce((acum, num) => {
+              typeof num[options[key]] === 'number' && counter++;
```

Comment:
use if instead, in this case(when you make mutation), this syntax is hard to read and maintain

Reply:
do not access external variables inside reduce callback